### PR TITLE
Allow pulling all nonlocal 'a' coefficient data

### LINKED ISSUE #24453: Add ability to sync all nonlocal 'a' coefficient data for Rhie-Chow
## Reason
In https://github.com/idaholab/moose/discussions/24093 and related issue https://github.com/ABallisat/CICSAM/issues/2 the user wants to be able to access the Rhie-Chow velocity beyond its natural stencil for testing different advection discretizations.

## Design
Add a `_pull_all_nonlocal` data member and associated API to `INSFVRhieChowInterpolator`

## Impact
Allow more exotic discretization exploration

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #24453

### CODE REVIEW

File: modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C [hunk 1]
```diff
@@ -439,14 +440,28 @@ INSFVRhieChowInterpolator::finalize()
...
     mooseAssert(it != _a.end(), "We definitely should have found something");
     push_data[pid].push_back(std::make_pair(id, it->second));
-    pull_requests[pid].push_back(id);
+  }
+
+  // Create pull data
+  if (_pull_all_nonlocal)
+  {
+    for (const auto * const elem :
+         as_range(_mesh.active_not_local_elements_begin(), _mesh.active_not_local_elements_end()))
```
Comment (GiudGiud) [inline modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:457]: I m curious how we know an element is active if it s not local

same how do we know their processor id or the elem id?

Comment (lindsayad) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:457]: If any process has an element then libMesh ensures that the data in that element is complete and uniform across all processes

Comment (GiudGiud) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:457]: even if the mesh is distributed?

Comment (lindsayad) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:457]: correct although in that case many processes may not have a given element at all

Comment (GiudGiud) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:457]: so the process does not have it, so it doesnt show up in the "non_local" list?

feels like "non_local" is  more "replicated_non_local"

Comment (lindsayad) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:457]: To me its implicit in `elements` that the range will not have null/nonexistent elements

Comment (lindsayad) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:457]: And `ghosted_non_local` means the same to me as `non_local`

Comment (lindsayad) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:457]: But you can debate this name with @roystgnr haha

Comment (roystgnr) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:457]: That name is kinda grandfathered in.  Added by John on Sep 7 2007; I started "ParallelMesh" a couple weeks later.

Comment (GiudGiud) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:457]: > And ghosted_non_local means the same to me as non_local

non_local to me could non_local and (ghosted or non_ghosted). Just not local.

**Review by GiudGiud — CHANGES_REQUESTED** [review]
- needs testing

- needs to be usable from the input file (so like a new parameter to the RC UO?)

- ideally, on a ghosting error on a coefficient, we would also tell a user to enable this.

**Review by GiudGiud — APPROVED** [review]

**Comment by lindsayad** [comment]
Thank you sir for the review!

**Pull request merged**